mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -1570,23 +1573,26 @@ private ScheduleRequest createSchedulingRequest(
                     }
                 }
 
+                JobMetadata jobMetadata = new JobMetadata(
+                    mantisJobMetaData.getJobId().getId(),
+                    mantisJobMetaData.getJobJarUrl(),
+                    mantisJobMetaData.getTotalStages(),
+                    mantisJobMetaData.getUser(),
+                    mantisJobMetaData.getSchedulingInfo(),
+                    mantisJobMetaData.getParameters(),
+                    getSubscriptionTimeoutSecs(mantisJobMetaData),
+                    getHeartbeatIntervalSecs(mantisJobMetaData),
+                    mantisJobMetaData.getMinRuntimeSecs()
+                );
                 ScheduleRequest sr = new ScheduleRequest(
                         workerId,
                         workerRequest.getStageNum(),
                         workerRequest.getNumberOfPorts(),
-                        new JobMetadata(
-                                mantisJobMetaData.getJobId().getId(),
-                                mantisJobMetaData.getJobJarUrl(),
-                                mantisJobMetaData.getTotalStages(),
-                                mantisJobMetaData.getUser(),
-                                mantisJobMetaData.getSchedulingInfo(),
-                                mantisJobMetaData.getParameters(),
-                                getSubscriptionTimeoutSecs(mantisJobMetaData),
-                                getHeartbeatIntervalSecs(mantisJobMetaData),
-                                mantisJobMetaData.getMinRuntimeSecs()
-                        ),
+                        jobMetadata,
                         mantisJobMetaData.getSla().orElse(new JobSla.Builder().build()).getDurationType(),
-                        stageMetadata.getMachineDefinition(),
+                        SchedulingConstraints.of(
+                            stageMetadata.getMachineDefinition(),
+                            mergeJobDefAndArtifactAssigmentAttributes(jobMetadata.getJobArtifact())),
```

Comment:
I would suggest making this a property of JobStageMetadata to semantically indicate the scheduling constraints the stage imposes upon. 
```
stageMetadata.getSchedulingConstraints()
```

Reply:
Good suggestion, I completely agree. However, this change would require significant adjustments to core interfaces like `IMantisStageMetadata`, which could complicate things. For now, I will add a TODO and I'll come back once the rest has been dealt with.